# Schedule
## Biology 364/664 Spring 2023

| Week:Date | Topic | Required Reading | Suggested | Project |
| :--- | ------ | -------- | -------- | ----- |
| w01:19Jan2023 | Using R & Rstudio | **ASR1** | PRP4-5 | Homework 01 |
| w02:24Jan2023 | Simple Data Exploration & Git | **ASR2-3** | PRP6-10 EDA4-7 OIS1-4  | Homework 02 | 
| w03:31Jan2023 | Data Visualization & Wrangling | **ASR4** | EDA15-16 OIS5-7 21st6-7 | Homework 03 |
| w04:07Feb2023 | Advanced Hypothesis Testing | **ASR5** **EDA5** | OIS8-9 21st8-9  | **R Tutorial Project** |
| w05:14Feb2023 | Models | **ASR6** **ASA.pdf** **Moving_beyond.pdf** | R4DS22-25 21st10-13 21st26-27 | Homework 05 |
| w06:21Feb2023 | Data Science Ethics | **Fraser.pdf** **MDSR-8**| PRP12 EDA12-14 21st16-21 | Project 01 Due | 
| w07:28Feb2023 | Questionable Data Practices | **ASR7** **Forstmeier.pdf** **Errington.pdf** | | Homework 07 | 
| w08:07Mar2023 | Best Practices in Data Science | **ASR9** | ASR8 | TAKEHOME MIDTERM |
| SPRING BREAK |
| w09:21Mar2023 | Transparency & Reproducibility | [RNA-seqlopedia](https://rnaseq.uoregon.edu/) | | **QRP Case Studies Project** Homework 09 |
| w10:28Mar2023 | Transcriptomics Intro | **DEBrowser.pdf** | 21st32 | Homework 10 |
| w11:04Apr2023 | Transcriptomics Project | **Williams.pdf** | | **Transcriptomics Pipeline Project** |
| w12:11Apr2023 | Transcriptomics Project | [ARRIVE 2.0](https://arriveguidelines.org/arrive-guidelines) | | Big Data Visualization |
| w13-14:18Apr2023 | Transcriptomics Project | **minitufte.pdf** |  | ARRIVE Checklist **Transcriptomics Presentations** |
| Finals Week | **14:45 EDT Monday, May 8** | Deadline |  | TAKEHOME FINAL |

## Textbooks

### Required Readings

**ASR**: Applied Statistics with R: A Practical Guide for the Life Sciences by Justin C. Touchon, https://oxford.universitypressscholarship.com/view/10.1093/oso/9780198869979.001.0001/oso-9780198869979

**EDA**: Exploratory Data Analysis, by Roger D. Peng, https://leanpub.com/exdata

**MDSR**: Modern Data Science with R, by Benjamin S. Baumer, Daniel T. Kaplan, and Nicholas J. Horton, https://mdsr-book.github.io/mdsr2e/

### Suggested Readings

**21st**: Statistical Thinking for the 21st Century, by Russell A. Poldrack, https://statsthinking21.github.io/statsthinking21-R-site/

**Hart**: A Primer in Biological Data Analysis and Visualization Using R, by Gregg Hartvigsen, ISBN: 9780231166997

**HBS**: Handbook of Biological Statistics, by John H. McDonald, http://www.biostathandbook.com/

**OIS**: Open Intro Statistics 4th Edition, https://leanpub.com/openintro-statistics

**PRP**: R Programming for Data Science, by Roger D. Peng, https://leanpub.com/rprogramming

**R4DS**: R for Data Science, by Garrett Grolemund and Hadley Wickham, https://r4ds.had.co.nz/

***

## Helpful Links

Setting up R Studio, Git, and Github:
https://happygitwithr.com/index.html

Troubleshooting Git:
https://bookdown.org/rdpeng/RProgDA/version-control-and-github.html

Tufte's Ten Rules of Data Design: 
http://www.sealthreinhold.com/school/tuftes-rules/index.php

Data to Viz R Graph Gallery: 
https://www.data-to-viz.com/

R Tutorials from Biology 364:
https://github.com/Bucknell-Biol364

Common statistical tests are linear models: 
https://lindeloev.github.io/tests-as-linear/

Choosing the correct statistical test: 
https://stats.idre.ucla.edu/other/mult-pkg/whatstat/

Making Statistics Intuitive: 
https://statisticsbyjim.com/

Scientists rise up against statistical significance: 
https://www.nature.com/articles/d41586-019-00857-9

Explaining PCA to your family 
https://stats.stackexchange.com/questions/2691/making-sense-of-principal-component-analysis-eigenvectors-eigenvalues
